spec.bs

```diff
@@ -8372,48 +8759,80 @@ To <dfn>batch and fetch trusted scoring signals</dfn> given a [=trusted scoring
+        1. [=list/For each=] |entry| in |entriesToBatch|:
+          1. Set |entry|'s [=trusted scoring signals request/reply=] to |result|.
+    1. Otherwise:
+      1. Select a non-[=list/is empty|empty=] [=set/subset=] of |batcher|'s [=trusted scoring
+        signals batcher/request map=][|key|], called |entriesToBatch|.
+      1. Set |entriesToBatch| to |batcher|'s [=trusted scoring signals batcher/request map=][|key|].
+      1. [=map/Remove=] |key| from |batcher|'s [=trusted scoring signals batcher/request map=].
+      1. Let « |requestBody|, |renderUrlIdMap|, |context| » be the result of [=building batched
+        trusted key value scoring signals request body=] for |entriesToBatch|.
+      1. Let |baseUrl| be |entriesToBatch|[0]'s [=trusted scoring signals request/base url=].
       1. Let |seller| be |entriesToBatch|[0]'s [=trusted scoring signals request/seller=].
       1. Let |scriptFetcher| be |entriesToBatch|[0]'s [=trusted scoring signals request/seller
         script fetcher=].
-      1. Let |policyContainer| be |entriesToBatch|[0]'s [=trusted scoring signals request/policy
-        container=].
-      1. If |fullSignalsUrl|'s [=url/origin=] is not [=same origin=] with |seller| then:
+      1. If |baseUrl|'s [=url/origin=] is not [=same origin=] with |seller| then:
         1. Let |allowCrossOriginTrustedScoringSignalsFrom| be the result of [=wait for cross origin
           trusted scoring signals authorization from a fetcher=] given |scriptFetcher|.
         1. If |allowCrossOriginTrustedScoringSignalsFrom| does not [=list/contain=]
-          |fullSignalsUrl|'s [=url/origin=], set |fullSignalsUrl| to null.
+          |baseUrl|'s [=url/origin=], set |baseUrl| to null.
       1. Let |result| be failure.
-      1. If |fullSignalsUrl| is not null:
-        1. Let «|allTrustedScoringSignals|, <var ignore>ignored</var>, |scoringDataVersion|» be
-          the result of [=fetching trusted signals=] with |fullSignalsUrl|, |seller|,
-          |policyContainer|, and false.
+      1. If |baseUrl| and |requestBody| are not null:
+        1. Let «|allTrustedScoringSignals|, <var ignore>ignored</var>, |scoringDataVersionMap|» be
+          the result of [=fetching trusted key value signals=] with |baseUrl|, |requestBody|,
+          |context|, |seller|, |entriesToBatch|[0]'s [=trusted scoring signals request/policy container=],
+          |renderUrlIdMap| and false.
         1. If |allTrustedScoringSignals| is an [=ordered map=]:
           1. Set |result| to a new [=trusted scoring signals reply=]
           1. Set |result|'s [=trusted scoring signals reply/all trusted scoring signals=] to
             |allTrustedScoringSignals|.
-          1. Set |result|'s [=trusted scoring signals reply/data version=] to |scoringDataVersion|.
+          1. [=list/For each=] |entry| in |entriesToBatch|:
+            1. Set |result|'s [=trusted scoring signals reply/data version=] to |scoringDataVersionMap|

```

Comment:
after another look, it still does not seem right. |data version| needs to be bound to |entry| based on what we discussed, not |result| then, because every |entry| is storing the same |result|. Here after this loop, you set |result|'s |data version| to the last |entry|'s |data version|